Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -613,148 +613,140 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
     // MARK: - App Messages
 
     // swiftlint:disable:next cyclomatic_complexity
-    public override func handleAppMessage(_ messageData: Data, completionHandler: ((Data?) -> Void)? = nil) {
+    public override func handleAppMessage(_ messageData: Data) async -> Data? {
```

Comment:
This is the gist of it... by making everything `async` with a return value, the handlers cannot avoid returning something.

With a closure the handlers could "forget" to execute the completion callback.